src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -352,15 +387,50 @@ public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReference
                 if (!docsGenerated) this.LogAndUpdate(ref this.CompilationStatus.Documentation, ErrorCode.DocGenerationFailed, Enumerable.Empty<string>());
             }
 
-            // invoking the given targets
+            // invoking rewrite steps in external dlls
 
-            foreach (var buildTarget in this.Config.Targets ?? ImmutableDictionary<string, BuildTarget>.Empty)
+            if (this.CompilationOutput != null)
             {
-                this.CompilationStatus.BuildTargets[buildTarget.Key] = 0;
-                var succeeded = this.PathToCompiledBinary != null && buildTarget.Value != null &&
-                    buildTarget.Value(this.PathToCompiledBinary, ex => this.LogAndUpdate(buildTarget.Key, ex));
-                if (!succeeded) this.LogAndUpdate(buildTarget.Key, ErrorCode.TargetExecutionFailed, new[] { buildTarget.Key });
-            } 
+                for (int i = 0; i < this.ExternalRewriteSteps.Length; i++)
+                {
+                    this.CompilationStatus.LoadedRewriteSteps[i] = this.ExecuteRewriteStep(this.ExternalRewriteSteps[i], this.CompilationOutput, out this.CompilationOutput);
```

Comment:
The way it is set up now the CompilationOutput may actually be modified even though the rewrite step failed. Could we please keep it so that the rewrite steps either succeeds, or does not modify the CompilationOutput.